src/ZProc.ml

```diff
@@ -12,6 +12,17 @@ type t = {
 
 type model = (string * Value.t) list
 
+let rec print_list = function
+[] -> ()
+| e::l -> print_string e ; print_string " " ; print_list l
+
+  
+let write_to_z3_query_file str is_append =
+  let filename = "smt_queries.smt2" in (
+  let outc =Out_channel.create ~append:is_append filename in
+  protect ~f:(fun ()->fprintf outc "%s\n" str)
+  ~finally:(fun() ->Out_channel.close outc))
```

Comment:
Recording all z3 queries this way is extremely expensive.

1) We should enable this recording only for debug builds, not normal builds.
2) We shouldn't be opening and closing a file again and again for each query. We should have an open handle and close it at the end of execution.

Look at the `Log` module which achieves both these goals. We should remove this function (and all calls) from this PR, and we can have a separate PR to improve the debug output which adds this functionality.

Reply:
I agree I will remove the z3 query logging for this PR